testing/validator/src/main/java/com/amazon/aoc/validators/CWMetricValidator.java

```diff
@@ -103,16 +105,27 @@ public void validate() throws Exception {
           }
 
           List<Metric> actualMetricList = Lists.newArrayList();
-          addMetrics(
-              CloudWatchService.SERVICE_DIMENSION,
-              serviceNames,
-              expectedMetricList,
-              actualMetricList);
-          addMetrics(
-              CloudWatchService.REMOTE_SERVICE_DIMENSION,
-              remoteServiceNames,
-              expectedMetricList,
-              actualMetricList);
+
+          // Add sets of dimesion filters to use for each query to CloudWatch
+          List<List<Pair<String, String>>> dimensionLists = Lists.newArrayList();
+          for (String serviceName : serviceNames) {
+            dimensionLists.add(
+                Arrays.asList(new Pair<>(CloudWatchService.SERVICE_DIMENSION, serviceName)));
+          }
+          for (String remoteServiceName : remoteServiceNames) {
+            dimensionLists.add(
+                Arrays.asList(
+                    new Pair<>(CloudWatchService.REMOTE_SERVICE_DIMENSION, remoteServiceName)));
+          }
+          dimensionLists.add(
+              Arrays.asList(
+                  new Pair<>(CloudWatchService.REMOTE_SERVICE_DIMENSION, "AWS.SDK.S3"),
+                  new Pair<>(CloudWatchService.REMOTE_TARGET_DIMENSION, "e2e-test-bucket-name")));
```

Comment:
Once the metric exists, how do we know whether the test that just occurred is the one that populated it or if it simply exists from a previous test run? It's entirely possible that a test from 2 hours ago created the metric and then all tests since then did not appropriately populate this metric but because of our `.withRecentlyActive("PT3H")` we are not finding out at all that this metric is flakily populated.

This is the same issue we had with trying to check `[RemoteService]` when it is equal to `["www.amazon.com"]` for example. (There are also other considerations for this aggregation which I'll mention in another comment)

Note: `withRecentlyActive()` only accepts PT3H